openerp/tools/translate.py

```diff
@@ -305,12 +305,16 @@ def html_translate(callback, value):
     if not value:
         return value
 
-    parser = etree.HTMLParser(encoding='utf-8')
-    trans = XMLTranslator(callback, 'html', parser)
-    wrapped = "<div>%s</div>" % encode(value)
-    root = etree.fromstring(wrapped, parser)
-    trans.process(root[0][0])               # html > body > div
-    return trans.get_done()[5:-6]           # remove tags <div> and </div>
+    try:
+        parser = etree.HTMLParser(encoding='utf-8')
+        trans = XMLTranslator(callback, 'html', parser)
+        wrapped = "<div>%s</div>" % encode(value)
+        root = etree.fromstring(wrapped, parser)
+        trans.process(root[0][0])               # html > body > div
+        value = trans.get_done()[5:-6]           # remove tags <div> and </div>
+    except Exception:
```

Comment:
My concern is that this translation function is hardwired into [`read_from_database()` ](https://github.com/odoo/odoo/blob/5bd0b05c22388d9f8b2082ade1131f06b2998c96/openerp/models.py#L3376-L3378). It breaks the CRUD layer of the framework when it fails for any reason - a critical problem.
Being unable to parse the source/translation of an HTML field is an internal error that is not fatal, akin to not finding a matching entry in `ir_translation`. At least that's as far as the end-user should be concerned... We've seen tagsoup samples that we can't parse but can be rendered by the browser anyway.

Now if we are sure that `ValueError` is the only surprise we'll get, great. Otherwise the robustness of the CRUD layer might be more important than possibly silencing an odd exception from time to time. BTW `ValueError` is pretty generic :wink: